tests/test_other.py

```diff
@@ -9875,3 +9875,9 @@ def test_emranlib(self):
     run_process([PYTHON, EMRANLIB, 'liba.a'])
     output = run_process([shared.LLVM_NM, '--print-armap', 'liba.a'], stdout=PIPE).stdout
     self.assertContained('Archive map', output)
+
+  def test_piping_stdin(self):
+    create_test_file('temp.h', '#include <string>')
+    outputEcho = run_process([PYTHON, EMCC, '-x', 'cpp-output', '-dM', '-E', '-'], input="#include <string>", stdout=PIPE).stdout
```

Comment:
outputStdin?  (i'm not sure what echo means here.

Also this test it testing both reading from stding and compiling to stdout.   Would it make more sense to just test the stdin part here?   (i.e. write to a `-o output.txt`).   Outputing to stdout be tested elsewhere (and probably already is?)

Reply:
actually I had used echo command while testing it manually, stuck along. I'll change that.

In this Test, I am just making sure that output from using a temporary file and stdin remain the same, since they are only preprocessing.

Reply:
if we output it to a file, we'll need to read again, that's why compared it directly